src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/todo';
+import { useState } from 'react';
+
+type TodosProperties = {
+  todos: Todo[];
+  titleValue: string;
+  userValue: string;
+  isFormSubmited: boolean;
+};
+
+type HandleChangeTodosState =
+  | React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLSelectElement>;
+
+const todosProperties: TodosProperties = {
+  todos: todosFromServer,
+  titleValue: '',
+  userValue: '0',
+  isFormSubmited: false,
+};
 
 export const App = () => {
+  const [todosState, setTodosState] =
+    useState<TodosProperties>(todosProperties);
+  const { todos, titleValue, userValue, isFormSubmited } = todosState;
+
+  const handleSubmitForm = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!titleValue.length || userValue === '0') {
+      return setTodosState(prev => ({
+        ...prev,
+        isFormSubmited: true,
+      }));
+    }
+
+    const userIdSearch = usersFromServer.find(
+      user => user.name === userValue,
+    )?.id;
+
+    const newTodo: Todo = {
+      id: todos.length + 1,
+      title: titleValue,
+      completed: false,
+      userId: userIdSearch as number,
+    };
+
+    return setTodosState(prev => ({
+      ...prev,
+      todos: [...prev.todos, newTodo],
+      titleValue: '',
+      userValue: '0',
+      isFormSubmited: false,
+    }));
+  };
+
+  const handleChangeTodosState = (e: HandleChangeTodosState) => {
+    const { value, id } = e.target;
+
+    if (id === 'titleInput') {
+      return setTodosState(prev => ({ ...prev, titleValue: value }));
+    }
+
+    return setTodosState(prev => ({ ...prev, userValue: value }));
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmitForm}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title: </label>
+          <input
+            id="titleInput"
+            type="text"
+            data-cy="titleInput"
+            value={titleValue}
+            onChange={handleChangeTodosState}
+          />
```

Comment:
it should have a placeholder attribute 
![image](https://github.com/user-attachments/assets/b6a71710-d7e4-441a-99fb-768f1da663eb)